### PR TITLE
Feature: Full permissions flow for frontend

### PR DESCRIPTION
**Pull request by lonalynn**
This PR implements all of the frontend-based functionality for users to scrape app permissions.

- Full UI to preview and scroll through all 38 app permissions for the first 5 apps!
- Properly displays permissions per search based on "Show Permissions" toggle being switched on/off. 
- Separate file for permissions columns in 'constants' - there is now a "permissionColumns" file for columns when permissions are displayed, in addition to the normal "columns" file.

Additional, unrelated fixes:
- Bugfixed search button so that it properly passes search term to backend when clicked. Previously, it would not pass the search term to the backend, but would still run a query when clicked.

### CODE REVIEW

File: .gitignore
Comment (akaash11) [inline .gitignore:1]: We are currently not using Amplify. I don't think these changes are relevant to this PR. But it gitignore so its fine

**Review by akaash11 — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (lonalynn) [reply .gitignore:1]: Removed unnecessary parts of gitignore.

File: src/constants/urlConstants.js
[the remark this answers is not in the document]
Comment (lonalynn) [reply src/constants/urlConstants.js:1]: Updated URL for deployment.

**Review by lonalynn — COMMENTED** [review]
Amplify .gitignore removed & URL changed to deployment.

**Comment by lonalynn** [comment]
TODO from @akaash11's comments during today's meeting:

- Fix "remove this title prop" bug
- Truncate title text w/ellipses
- Info tooltip on "show permissions"

**Comment by lonalynn** [comment]
Fixed:
- "Remove this title prop" bug
- Info tooltip

TBD but most likely to be backlogged:
- Truncate titlte text with ellipses

**Review by akaash11 — APPROVED** [review]
Permissions tooltip is added.
And the console error is resolved.

**Pull request merged**